### PR TITLE
Changes that support new design change for adding recitations

### PR DESCRIPTION
**Pull request by achillesecos**
Parent: https://github.com/googleinterns/step78-2020/commit/b4a4c823021878370a236e12021dda8cebf4f576

Made the changes that will merge to `HandleRecitations`, the branch where everyones individual changes will merge into, so that the design change for Recitations is integrated properly, and can then be merged into `master`.

### CODE REVIEW

**Review by hjohns28 — APPROVED** [review]
LGTM

File: src/main/java/com/google/sps/data/ScheduledCourse.java
Comment (maceonthompson) [inline src/main/java/com/google/sps/data/ScheduledCourse.java:86]: It looks like this typo has since been fixed in the HandleRecitations branch (I'm guessing heidi also noticed it and I'm assuming fixed it in her PR). This means that you can actually rename it to the proper thing (`getLectureSection`), refactor the rest of your code accordingly, and it should work!

File: src/main/java/com/google/sps/data/RestrictTimesCriteria.java
Comment (maceonthompson) [inline src/main/java/com/google/sps/data/RestrictTimesCriteria.java]: Are you intending to not take into consideration a course's lab/recitation meeting times as well?

**Review by maceonthompson — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/sps/data/ScheduledCourse.java
Comment (achillesecos) [reply src/main/java/com/google/sps/data/ScheduledCourse.java:86]: Ok sounds good. I will change it to `getLectureSection()`

File: src/main/java/com/google/sps/data/RestrictTimesCriteria.java
Comment (achillesecos) [reply src/main/java/com/google/sps/data/RestrictTimesCriteria.java]: I will add that check, after I pull from the parent branch.

Comment (achillesecos) [reply src/main/java/com/google/sps/data/RestrictTimesCriteria.java]: Added the labSections [here](https://github.com/googleinterns/step78-2020/pull/33/commits/b444eee8391730e2d7056eb1f646cb40ebac30b6)

File: src/main/java/com/google/sps/data/PrioritizeCoursesCriteria.java
Comment (maceonthompson) [inline src/main/java/com/google/sps/data/PrioritizeCoursesCriteria.java]: `getlectureSections` --> `getLectureSections`

**Review by maceonthompson — APPROVED** [review]
LGTM pending the one change in PrioritizeCoursesCriteria!

File: src/main/java/com/google/sps/data/PrioritizeCoursesCriteria.java
Comment (achillesecos) [reply src/main/java/com/google/sps/data/PrioritizeCoursesCriteria.java]: Sounds good I made the change

File: src/test/java/com/google/sps/RankTest.java
Comment (FabriceCastel) [inline src/test/java/com/google/sps/RankTest.java:351]: Is there a reason we expect this specific sort order when there are no preferences? Should the list order end up getting changed at all when there are no preferences?

Comment (achillesecos) [reply src/test/java/com/google/sps/RankTest.java:351]: You're right, the list order should not change when there are no preferences set. I couldn't run tests anyways since it wont compile, until its accessible from `HandleRecitations` branch, so once its merged there, I will correct/add the rest of the tests.

**Review by FabriceCastel — CHANGES_REQUESTED** [review]
Marking the request as "requested changes" for now. Looks good, just needs the change following the other branch getting merged. Please re-open once the other branch is merged and you've made the matching updates here.

**Comment by achillesecos** [comment]
By the branch getting merged, do you mean `HandleRecitations` branch? I was planning on merging this PR, into the `HandleRecitations` so that I could make another PR from there that includes all the necessary change for the added tests to work.

**Review by FabriceCastel — APPROVED** [review]
LGTM

**Pull request merged**
